src/main/php/lang/GenericTypes.class.php

```diff
@@ -173,25 +173,35 @@ public function newType0($base, $arguments) {
           }
         } else if (2 === $state[0]) {             // Method declaration
           if ('(' === $tokens[$i][0]) {
+            if (0 === $braces) {
+              $src.= ' '.$m.'(';
+            } else {
+              $default[$p].= '(';
+            }
             $braces++;
           } else if (')' === $tokens[$i][0]) {
             $braces--;
             if (0 === $braces) {
-              array_shift($state);
-              $src.= ')';
-              continue;
+              $src.= isset($default[$p]) ? '= '.$default[$p].')' : ')';
+            } else {
+              $default[$p].= ')';
             }
-          }
-          if (T_VARIABLE === $tokens[$i][0]) {
+          } else if ('{' === $tokens[$i][0] || ';' === $tokens[$i][0]) {
+            array_shift($state);
+            array_unshift($state, 3);
+            $i--;
+          } else if (T_VARIABLE === $tokens[$i][0]) {
             $parameters[]= $tokens[$i][1];
+            $src.= $tokens[$i][1];
           } else if (',' === $tokens[$i][0]) {
-            // Skip
+            $src.= (isset($default[$p]) ? '= '.$default[$p].',' : ',');
+            $p++;
           } else if ('=' === $tokens[$i][0]) {
-            $p= sizeof($parameters)- 1;
             $default[$p]= '';
           } else if (T_WHITESPACE !== $tokens[$i][0] && isset($default[$p])) {
             $default[$p].= is_array($tokens[$i]) ? $tokens[$i][1] : $tokens[$i];
           }
+          continue;
         } else if (3 === $state[0]) {             // Method body
           if (';' === $tokens[$i][0]) {
             // Abstract method
```

Comment:
The changes in this file create wrapper methods without type hints. Not sure if this is always desirable!